### DISCUSSION TITLE
Using https requests instead of websockets

### DISCUSSION THREAD

**Discussion by GSri30** [TypeFox/monaco-languageclient#504] [Q&A] [answered]
Hello,

I'm exploring the `monaco-languageclient` package. 

*Requirement:* To connect a monaco-editor (react app) with a language server provided by xtext via servlets (Completely use HTTP requests instead of websocket connection).

*Progress:* Currently exploring these files
(1) https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/main/src/common.ts
(2) https://github.com/TypeFox/monaco-languageclient/tree/main/packages/examples/main/src/react

There is an implementation of connecting via websockets. I was wondering if it is possible to eliminate the websockets and tune the code to make it work via https requests. 

*Doubts:*
(1) Is it even possible to achieve that?
(2) If so, what could be the possible flow to follow (I'm completely new to this dimension)

**Comment by kaisalmen** [comment, marked as answer]
Hi @GSri30 you have to implement it yourself.  But you can just write someting the implements the `MessageReader` and `MessageWriter` interfaces. We currently use two different ones in our examples:
**jsonrpc over websocket**:
https://github.com/TypeFox/monaco-languageclient/blob/main/packages/vscode-ws-jsonrpc/src/socket/reader.ts
https://github.com/TypeFox/monaco-languageclient/blob/main/packages/vscode-ws-jsonrpc/src/socket/writer.ts
used here:
https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/main/src/common.ts#L44-L58

and **web worker messages**:
https://github.com/microsoft/vscode-languageserver-node/blob/main/jsonrpc/src/browser/main.ts (web worker)
used here:
https://github.com/TypeFox/monaco-languageclient/blob/main/packages/examples/main/src/langium/main.ts#L116-L118

So the way forward starts with implementing an `AbstractMessageReader` and `AbstractMessageWriter` yourself and plug those into `createLanguageClient`.

**Reply by GSri30** [reply]
Yeah got it. Thanks.
